### ISSUE TITLE
Support videos in markdown as plugin like images

### ISSUE THREAD

**Issue by josephshaaban** [django-wiki/django-wiki#1089]
I was just wondering why the videos weren't supported as markdown tag in wiki content in a way which maybe similar to that used in images plugin. Are you planning to support them in the next release or in the future release?

I'm using Django-wiki in my project and I have to support them. So If you are planning to support them in the next release I can help adding and supporting this new feature? Or I have to switch to another approach which is making a new extension provides this feature?

**Comment by benjaoming** [comment]
I would happily accept a PR based on the images plugin's structure. It could be called "videos". It's a great idea!

As there are many complications with videos regarding transcoding, resizing and such, it's not obvious exactly how to implement it in a user-friendly or easily accessible way. If this should be a bundled core plugin, I would suggest to keep the implementation very simple and have configurable options. For instance, the plugin should _not_ require django-wiki to access ffmpeg by command line and such.

Another smaller scope could be to have a plugin that support different patters of "embed"-like code snippets.

**Comment by teury** [comment]
@josephshaaban 
https://pyembed.github.io/

pyembed-markdown

you can use this package it is very easy to implement

**Comment by benjaoming** [comment]
@josephshaaban did you start implementing anything or have time to check out @teury's suggestions?

**Comment by josephshaaban** [comment]
Yes, I started implementing a demo on my local machine without transcoding which allow user upload mp4 files only (I know it's not the best practice, but HTML5 doesn't allow you to use video tag without determine a specific type for the video source (also there is a way to solve this, but for our demo I think it's enough)).

The next two steps are:

- Add transcoding feature (This will take some time to be finished and supported).
- Support embedding online video from (Youtube, Vimeo, Dailymotion, Yahoo, Veoh, & Metacafe) in same plugin.

I'm working on this plugin as part time, so my progress on this feature is a bit slow.

Would you like to add a new branch for my demo and keep you up-to-date?

**Comment by benjaoming** [comment]
@josephshaaban - the embedding would perhaps be better suited for a separate plugin? This is definitely also easier to maintain as a core plugin, I imagine the feature set will be smaller and less complex.

Sounds super-promising!

You are welcome to open a PR, you can use the "Draft" feature to indicate that it's a work in progress.

**Comment by benjaoming** [comment]
Nice that you started the work! It looks very promising judging from https://github.com/django-wiki/django-wiki/compare/main...josephshaaban:django-wiki:1089-support-videos-as-plugin

LMK, if you would like inputs at this stage :+1:

**Comment by josephshaaban** [comment]
Thank you @benjaoming for the quick review!

If you have any inputs at this stage, I appreciate. In addition, I will open PR ASAP so if there is any comments or advices before openning the PR are more welcome.

**Comment by benjaoming** [comment]
This is great to hear! I think it looks promising, and I suppose that at this stage, the plugin doesn't have external dependencies in order to work?

Some comments:

* It's always easier to give feedback in a PR, as you can use comments and inline-suggestions

* There's a super-nice recipe to extend allowed tags for bleach here: https://github.com/django-wiki/django-wiki/pull/1316/files#diff-150926d1058985e6603c32e0b34c4b1444a85cecc4861b9cd5d4082c0d3c5bc8R17 - so instead of extending the tag list immediate, it's only extended when the plugin is enabled.

* TESTS! Really, this needs some level of testing so things will not suddenly break.
